imblearn/scaling/css.py

```diff
@@ -0,0 +1,249 @@
+        Returns
+        -------
+        self : object,
+            Return self.
+
+        """
+
+        super(CSS, self).fit(X, y)
+
+        self._validate_estimator()
+
+        return self
+
+    def _shuffleTwo(self, a, b):
+        #if len(a) != len(b):
+        #    raise ValueError("lenth of a ({}) doesn't match length of b ({})".format(len(a), len(b)))
+
+        indexes = np.array(range(0, len(a)))
+        random.shuffle(indexes)
+        a2, b2 = a[indexes], b[indexes]
+
+        return a2, b2, indexes
+
+    def _sample(self, X, y):
+        """scales the dataset.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_samples, n_features)
+            Matrix containing the data which have to be sampled.
+
+        y : ndarray, shape (n_samples, )
+            Corresponding label for each sample in X.
+
+        Returns
+        -------
+        X_scaled : ndarray, shape (n_samples, n_features)
+            The array containing the resampled data.
+
+        y_scaled : ndarray, shape (n_samples)
+            The corresponding label of `X_scaled`
+
+        """
+
+        if self.mode not in CSS_MODE:
+            raise ValueError('Unknown kind for CSS mode.'
+                             ' Choices are {}. Got \'{}\' instead.'.format(
+                            CSS_MODE, self.mode))
+
+        if self.target not in CSS_TARGET:
+            raise ValueError('Unknown kind for CSS target.'
+                             ' Choices are {}. Got \'{}\' instead.'.format(
+                                CSS_TARGET, self.target))
+
+        if self.c < 0 or self.c > 1:
+            raise ValueError('Received scaling factor c={}, which'
+                             ' is outside the allowed range '
+                             '(0-1].'.format(self.c))
+        if self.c is 0:
+            raise ValueError('Received scaling factor c={}, which is'
+                             ' equal to no CSS at.'.format(self.c))
+
+        if self.minority_class_value is not None and \
+                not isinstance(self.minority_class_value, int):
+            raise ValueError('Unallowed target class value \'{}\'.'
+                             ' Valid values include None to automatically'
+                             ' infer the target class or any integer number'
+                             ' corresponding to the value of the label in y')
+
+
+        mcv = self.minority_class_value
+        if mcv is None:
+            # infer minority class value
+            counts = Counter(y)
+            least_common = counts.most_common()[:-1-1:-1]
+            mcv = least_common[0][0]
+
+        # in the following _a is majority, _i is minority
+        if self.target is "majority" or self.target is "both":
+            col_means_a = np.mean(X[(y != mcv)], axis=0)
+            if self.mode is "linear":
+                distances_a = abs(np.subtract(X[y != mcv], col_means_a))
+        if self.target is "minority" or self.target is "both":
```

Comment:
we need to extend the problem to multiclass

Reply:
right now, it only works in a 1-vs-all fashion. Is this required for an accept ?

Reply:
Whatever was shown in the paper is fine. This is usually the way that people transform the balancing method for multiclass. The only thing is that the `sampling_strategy` (which will replace `target`` should a string which can be (auto, all, minority, majority, not minority, not majority). In short by calling the BaseSampler it will return a `sampling_strategy` which is a dict with the class that should be scaled depending of the string. Then you scaling should loop other the key.

I imagine that we should also accept a list which mention the classes which should be scaled.

But anyway, we need to merge #413 first. So I would address all other issue.